lib/jwt/jwk/factory.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module JWT
+  module JWK
+    class Factory
```

Comment:
Is this a Factory or just a Base?

Otherwise this looks good. We could then refactor the other classes to use this when they reach master.

Reply:
It is an Abstract! What was I thinking here? So, we need to rename `Factory` to some other name. Suggestions?

Reply:
Maybe something with `key` as the RFC states "structure that represents a cryptographic key". 

`:: JWT::JWK::KeyBase` or `:: JWT::JWK::KeyAbstract` would be my choices. I have no better suggestions :)

Reply:
Thx! I commit the changes with `JWT::JWK::KeyAbstract`.